packages/core-app-api/src/app/AppManager.tsx

```diff
@@ -154,7 +154,53 @@ function useConfigLoader(
     };
   }
 
-  const configReader = ConfigReader.fromConfigs(config.value ?? []);
+  let configReader;
+
+  /**
+   * config.value can be undefined or empty. If it's either, don't bother overriding anything.
+   */
+  if (config.value?.length) {
+    configReader = ConfigReader.fromConfigs(config.value);
+
+    const resolveRelativeUrl = (relativeUrl: string) =>
+      new URL(relativeUrl, document.location.origin).href;
+
+    const getRelativeUrl = (fullUrl: string) => {
+      const url = new URL(fullUrl);
+      return fullUrl.replace(url.origin, '');
+    };
+
+    const getOrigin = (url: string) => new URL(url).origin;
+
+    const appBaseUrl = configReader.getString('app.baseUrl');
+    const backendBaseUrl = configReader.getString('backend.baseUrl');
+    const appOrigin = getOrigin(appBaseUrl);
+    const backendOrigin = getOrigin(backendBaseUrl);
+
+    /**
+     * We only want to override the URLs with the document origin when the URLs match
+     *  and are defined. We use getOptionalString here to not throw when the app.baseUrl
+     *  and backend.baseUrl are not defined. If they are defined but not well formatted URLs
+     *  the above getRelativeUrl() method will throw.
+     */
+    if (appOrigin === backendOrigin) {
+      config.value.push({
```

Comment:
Let's avoid mutating `config.value` and instead concat the new config in. Tbh I think it's best if we avoid adding this config if it doesn't change anything, so feeling it's best to structure this logic similar to what you had in the CLI before.